sos/plugins/__init__.py

```diff
@@ -323,7 +323,12 @@ def do_file_sub(self, srcpath, regexp, subst):
                 replacements = 0
         except Exception as e:
             msg = "regex substitution failed for '%s' with: '%s'"
-            self._log_error(msg % (path, e))
+            # if trying to regexp a nonexisting file, dont log it as an
+            # error to stdout
+            if isinstance(e, FileNotFoundError):
```

Comment:
Could it be better to test for `e.errno == errno.ENOENT`? That way we would still do the right thing if the class isn't actually an instance of `FileNotFound`. We're in standard library modules here, so perhaps we can already conclude it's guaranteed to only be this class but possibly worth checking (we use `errno` in a number of other places for this discrimination).

Reply:
Sure, PR rebased.